sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ParDoSchemaTest.java

```diff
@@ -388,7 +388,7 @@ public void testInferredSchemaPipeline() {
                     new DoFn<Inferred, String>() {
                       @ProcessElement
                       public void process(@Element Row row, OutputReceiver<String> r) {
-                        r.output(row.getString(0) + ":" + row.getInt32(1));
+                        r.output(row.getString("stringField") + ":" + row.getInt32("integerField"));
```

Comment:
So basically it is not safe to use AutoValue with Schemas. Should we convert the example to use a Pojo instead (like it previously did)? Should we throw an error that schemas do not work deterministically in this case?

Reply:
It is perfectly safe. What's not safe is assuming that the order of the schema fields will match the order of the methods in your class. In fact it's technically not safe for POJOs either, as AFAICT Java makes no guarantees there either.

Reply:
I just wondered whether we should throw an error when users try to access fields via a non-deterministic position? It seems like this could cause subtle bugs when two fields have the same type.